src/Fields/fieldvector.jl

```diff
@@ -184,6 +185,43 @@ end
     return dest
 end
 
+"""
+    Spaces.create_dss_buffer(fv::FieldVector)
+
+Create a NamedTuple of buffers for communicating neighbour information of
+each Field in `fv`. In this NamedTuple, the name of each field is mapped
+to the buffer.
+"""
+function Spaces.create_dss_buffer(fv::FieldVector)
+    NamedTuple{propertynames(fv)}(
+        unrolled_map(
+            key -> Spaces.create_dss_buffer(getproperty(fv, key)),
+            propertynames(fv),
+        ),
+    )
+end
+
+"""
+    Spaces.weighted_dss!(fv::FieldVector, dss_buffer = Spaces.create_dss_buffer(fv))
+
+Apply weighted direct stiffness summation (DSS) to each field in `fv`.
+Reuse the same `dss_buffer` for all fields in `fv`, which is constructed using
+the first field in `fv` if it isn't passed explicitly.
+"""
+# TODO distribute `propertynames(fv)` over processes to parallelize `unrolled_foreach`
```

Comment:
Could you move this comment to the end of the function? Putting it before the function will probably mess up the docstring.